### PR TITLE
Add PacketV2 Protos and CommitmentV2 function

### PR DESCRIPTION
**Pull request by chatton**
## Description

This PR adds the PacketV2 protos which we will use internally to convert a PacketV1 to a PacketV2 and commit based on the protocol version field.

I think the final hashing function and structure is not 100% set in stone yet, and might still change based on how the implementation goes.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: proto/ibc/core/channel/v1/channel.proto [hunk 1]
```diff
@@ -174,6 +174,42 @@ message PacketId {
...
+  // number corresponds to the order of sends and receives, where a Packet
+  // with an earlier sequence number must be sent and received before a Packet
+  // with a later sequence number.
+  uint64 sequence = 1;
+  // identifies the sending chain.
+  string source_id = 2;
+  // identifies the receiving chain.
+  string destination_id = 3;
+  // timeout after which the packet times out
+  string timeout = 4;
```
Comment (chatton) [inline proto/ibc/core/channel/v1/channel.proto:188]: not convinced this is the best idea, @damiannolan also pointed out that string parsing is expensive in an EVM environment. 

Is there a good reason not to just use uint64 nanos only?

cc @AdityaSripal

Comment (AdityaSripal) [reply proto/ibc/core/channel/v1/channel.proto:188]: I also agree that we should just have uint64

File: modules/core/04-channel/types/packet.go [hunk 2]
```diff
@@ -100,6 +101,43 @@ func commitV2Packet(packet Packet) []byte {
 	return hash[:]
 }
 
+// CommitPacketV2 returns the V2 packet commitment bytes. The commitment consists of:
+// sha256_hash(timeout) + sha256_hash(destinationID) + sha256_hash(packetData) from a given packet.
+// This results in a fixed length preimage.
+// NOTE: A fixed length preimage is ESSENTIAL to prevent relayers from being able
+// to malleate the packet fields and create a commitment hash that matches the original packet.
+func CommitPacketV2(packet PacketV2) []byte {
```
[the remark this answers is not in the document]
Comment (chatton) [reply modules/core/04-channel/types/packet.go:109]: I think this'll end up being used in the `packet-server` package when acting on PacketV2 types and so will be public.

File: modules/core/04-channel/types/packet.go [hunk 3]
```diff
@@ -26,6 +26,7 @@ func CommitPacket(packet Packet) []byte {
 	case IBC_VERSION_UNSPECIFIED, IBC_VERSION_1:
 		return commitV1Packet(packet)
 	case IBC_VERSION_2:
+		// TODO: convert to PacketV2 and commit.
```
[the remark this answers is not in the document]
Comment (chatton) [reply modules/core/04-channel/types/packet.go:29]: don't think so, this'll just be handled as part of PacketV2 internal logic refactor. Can create an issue for this specifically if you think it's worth it!

File: modules/core/04-channel/types/packet.go [hunk 4]
```diff
@@ -100,6 +101,42 @@ func commitV2Packet(packet Packet) []byte {
 	return hash[:]
 }
 
+// CommitPacketV2 returns the V2 packet commitment bytes. The commitment consists of:
+// sha256_hash(timeout) + sha256_hash(destinationID) + sha256_hash(packetData) from a given packet.
```
Comment (crodriguezvega) [inline modules/core/04-channel/types/packet.go:105]:
```suggestion
// sha256_hash(timeout) + sha256_hash(destinationID) + sha256_hash(packetData) for a given packet.
```

File: modules/core/04-channel/types/packet.go [hunk 5]
```diff
@@ -26,6 +26,7 @@ func CommitPacket(packet Packet) []byte {
 	case IBC_VERSION_UNSPECIFIED, IBC_VERSION_1:
 		return commitV1Packet(packet)
```
Comment (crodriguezvega) [inline modules/core/04-channel/types/packet.go:27]: Can be of course done in a separate PR, but probably we can rename this to `commitPacketV1`?

**Review by crodriguezvega — APPROVED** [review]
Looks good. Thanks, @chatton!

File: modules/core/04-channel/types/packet.go (hunk 4 above)
Comment (chatton) [reply modules/core/04-channel/types/packet.go:105]: sorry missed this comment, will include this in a follow up!
